src/v/cloud_storage/remote_partition.cc

```diff
@@ -330,6 +330,9 @@ ss::future<> remote_partition::run_eviction_loop() {
         }
     } catch (const ss::broken_condition_variable&) {
     }
+    for (auto& rs : _eviction_list) {
+        co_await std::visit([](auto&& rs) { return rs->stop(); }, rs);
+    }
```

Comment:
not exactly, `wait` will throw `ss::broken_condition_variable` exception since we're calling `_cvar.broken()`
the exception will be caught and then we will stop the remaining elements inside the eviction list